packages/frontend/src/hooks/useLatestKPITokens.ts

```diff
@@ -25,20 +25,21 @@ export function useLatestKPITokens(limit = 5): {
             });
             if (kpiTokensAmount === 0) return [];
 
-            const fromIndex = Math.max(kpiTokensAmount - limit, 0);
-            const kpiTokenAddresses = await Fetcher.fetchKPITokenAddresses({
-                publicClient,
-                preferDecentralization,
-                fromIndex,
-                toIndex: kpiTokensAmount,
-            });
+            const kpiTokenAddresses =
+                await Fetcher.fetchLatestKPITokenAddresses({
+                    publicClient,
+                    preferDecentralization,
+                    limit,
+                });
             const kpiTokens = await Fetcher.fetchKPITokens({
                 publicClient,
                 preferDecentralization,
                 addresses: kpiTokenAddresses,
             });
 
-            return Object.values(kpiTokens).reverse();
+            return preferDecentralization
```

Comment:
Not much of a fan of this... I'd prefer if both the onchain and subgraph versions would return arrays ordered the same way. It's probably ok to return them ordered by creation time in asc order and then do a reverse here.